src/new_hope/week1.clj

```diff
@@ -21,8 +21,121 @@
 (facts "some facts"
   (second [2 3 4]) => 3)
 
+
+;; Day 3
+
+(re-seq #"jam" "I like jam in my jam")
+
+(apply str [1 2 3])
+
+(apply str (re-seq #"A-Z+" "AadfBdfasZaa"))
+
+(.toUpperCase (str (first [:cat :dog :fish])))
+
+(-> [:cat :dog :fish]
+    first
+    str
+    .toUpperCase)
+
+(->> [1 2 3 4 5 6 7 8]
+     (filter odd?)
+     (take 2))
+
+(for [x (range 40)
+        :when (= 1 (rem x 4))]
+  x)
+
+(for [x (iterate #(+ 4 %) 0)
+      :let [z (inc x)]
+      :while (< z 40)]
+  x)
+
+(for [[x y] (partition 2 (range 20))]
+  [x y])
+
 ;; Day 4
 
+(defn second-to-last [col]
+  (if (< (count col) 2)
+    nil
+    (nth col (- (count col) 2))))
+
+(facts "should return the second to last"
+       (second-to-last (list 1 2 3 4 5)) => 4
+       (second-to-last ["a" "b" "c"]) => "b"
+       (second-to-last [[1 2] [2 3]]) => [1 2]
+       (second-to-last []) => nil
+       (second-to-last [1]) => nil)
+
+(defn sum [col]
+  (reduce + col))
+
+(facts "should return the sum"
+       (sum [1 2 3]) => 6)
+
+(defn odd-numbers [col]
+  (filter odd? col))
+
+(facts "should return the odd numbers"
+       (odd-numbers #{1 2 3 4 5}) => '(1 3 5))
+
+(defn palindrome [col]
```

Comment:
Actually, a simpler solution exists, it's enough to compare the original sequence to its reversed version